docker/src/main/java/clocker/docker/location/DockerContainerLocation.java

```diff
@@ -269,13 +273,100 @@ private void parseDockerCallback(String commandString) {
     }
 
     @Override
+    public int copyTo(final Map<String,?> props, final InputStream src, final long filesize, final String destination) {
+        Map<String,?> nonPortProps = Maps.filterKeys(props, Predicates.not(Predicates.containsPattern("port")));
+        boolean entitySsh = Boolean.TRUE.equals(entity.config().get(DockerContainer.DOCKER_USE_SSH));
+        boolean dockerSsh = Boolean.TRUE.equals(getOwner().config().get(DockerContainer.DOCKER_USE_SSH));
+        if (entitySsh && dockerSsh) {
+            return super.copyTo(nonPortProps, src, filesize, destination);
+        } else {
+            return copyTo(props, src, destination);
+        }
+    }
+
+    @Override
+    public int copyTo(final Map<String,?> props, final InputStream src, final String destination) {
+        Map<String,?> nonPortProps = Maps.filterKeys(props, Predicates.not(Predicates.containsPattern("port")));
+        boolean entitySsh = Boolean.TRUE.equals(entity.config().get(DockerContainer.DOCKER_USE_SSH));
+        boolean dockerSsh = Boolean.TRUE.equals(getOwner().config().get(DockerContainer.DOCKER_USE_SSH));
+        if (entitySsh && dockerSsh) {
+            return super.copyTo(nonPortProps, src, destination);
+        } else {
+            File tmp = null;
+            try {
+                tmp = File.createTempFile(dockerContainer.getId(), Urls.getBasename(destination));
+                hostMachine.copyTo(nonPortProps, src, tmp.getAbsolutePath());
```

Comment:
That's how I read the code (except we have also create a temp file on the clocker server because that is what `File.createTempFile` does). I have two concerns:
1. the path to the tmp file that was sensible on the clocker server may be entirely unsuitable on the docker host, causing it to fail;
2. the file on the docker host is not a temporary file - nothing deletes it. That could be a security issue, depending on the nature of the file being copied.